crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -275,13 +297,48 @@ impl WorkerPool {
         Span::current().record("batch_id", proposed_batch.id().to_string());
         info!(target: COMPONENT, "Proposed Batch built");
 
-        let proven_batch = LocalBatchProver::new(MIN_PROOF_SECURITY_LEVEL)
-            .prove(proposed_batch)
-            .map_err(BuildBatchError::ProveBatchError)?;
+        let proven_batch = batch_prover.prove(proposed_batch).await?;
 
         Span::current().record("batch_id", proven_batch.id().to_string());
         info!(target: COMPONENT, "Proven Batch built");
 
         Ok(proven_batch)
     }
 }
+
+// BATCH PROVER
+// ================================================================================================
+
+/// Represents a batch prover which can be either local or remote.
+#[derive(Clone)]
+pub enum BatchProver {
+    Local(LocalBatchProver),
+    Remote(RemoteBatchProver),
+}
+
+impl BatchProver {
+    pub fn new_local(security_level: u32) -> Self {
+        info!(target: COMPONENT, "Using local batch prover");
+        Self::Local(LocalBatchProver::new(security_level))
+    }
+
+    pub fn new_remote(endpoint: impl Into<String>) -> Self {
+        info!(target: COMPONENT, "Using remote batch prover");
+        Self::Remote(RemoteBatchProver::new(endpoint))
+    }
```

Comment:
Not for this PR, but here we'll also need to provide required security level and then check if the proofs returned from the remote prover match the expected security level.